docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,271 @@
+      machineImages:
+        - name: suse-chost
+          versions:
+            - version: 16.4
+              expirationDate: 2023-08-8T23:59:59Z
+            - version: 15.4
+            - version: 14.4
+            - version: 13.6
+      machineTypes: 
+        - name: m5.large
+          cpu: "4"
+          gpu: "0"
+          memory: 8Gi
+        - name: m5.xlarge
+          cpu: "8"
+          gpu: "0"
+          memory: 16Gi
+      volumeTypes:
+        - name: ab6
+          class: premium
+          usable: true
+        - name: gp3
+          class: standard
+          usable: true
+```
+
+The rendering is done by a new custom controller registered to the `gardener-controller-manager`. Merge conflicts can not arise during the merge process as they are caught by static validation and an admission plugin for validating the `PrivateCloudProfile` object.
+
+### Custom RBAC verb
+
+To prevent users from entering arbitrary values in the `kubernetes` and `machineImages` fields, two custom RBAC verbs may be introduced. It can then be checked if the user that is creating or updating the `kubernetes` or `machineImages` field is authorized to do so.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: privatecloudprofile-modify-special-fields
+rules:
+  - apiGroups: [""]
+    resources: ["privatecloudprofile"]
+    verbs: ["modify-spec-kubernetes", "modify-spec-machineImages"]
+```
+
+### Automated removal of outdated Kubernetes versions and machine image versions
+
+Expired Kubernetes versions in `CloudProfile`s are currently being removed manually. This is not an issue since there is a very limited amount of `CloudProfile`s. However, if any project administrator is allowed to create a `PrivateCloudProfile` that can then have its Kubernetes versions expiration dates extended, it could become a very cumbersome process to manually remove all the expired Kubernetes versions. One possible solution is to implement a custom controller managed by `gardener-controller-manager`, that reconciles `PrivateCloudProfile`s (and possibly even regular `CloudProfile`s) and removes any Kubernetes versions that are past their expiration date. The same could be done for machine image versions.
+
+However, a problem arises as the Kubernetes/machine image versions are referenced in the `providerConfig` and would therefore need to be removed manually anyway. Because of that, this feature is not part of this GEP.
+
+### Cross-project sharing
+
+A use case could be defined where a `PrivateCloudProfile` might want to be shared across multiple projects and not just be used within the project it was created in. However, this use case seems to be very slim so this functionality will not be implemented as of now. However, when taking a broader view of Private Seeds and Cloud in Country, this feature is going to be necessary at some point. Still, it is not planned to be implemented as part of this GEP.
```

Comment:
I do not believe we have sufficient data to consider this use case "very slim". Where is the data coming from?

If we argue, without data, then let me do that as well: The whole use case of private cloud profiles is probably slim, but because we have a single level project hierarchy (no hierarchy) people share credentials, members, and configuration (in fact, that could be even proven or disproven).

This, forcing people to copy everything between accounts (in our case projects) was/is the biggest UX problem for a major platform I am not naming here.

Reply:
Clarified in private discussion.